cfgmgr/shellcmd.h

```diff
@@ -18,4 +21,10 @@
     }                                           \
 })
 
+static inline std::string shellquote(const std::string& str)
+{
+    static const std::regex re("([$`\"\\\n])");
+    return "\"" + std::regex_replace(str, re, "\\$1") + "\"";
```

Comment:
How can we be sure this regex is enough for everything?
Should we better create functions to check the input parameters and run them?
Our input parameters types are easy:
1. ip address
2. network mask
3. network device
4. number
5. string enum

Reply:
Update the PR comment at the top with answer.
I agree case-by-case protection is also possible, this one is for general purpose.